feat(marketing-analytics): add compareFilter for date range compariso…

jabahamondes opened the pull request:
## Changes

- Added comparison period support 
- Used columns render prop so I can display the tuple as the image
- Update the table css to show the two period values and information

## How did you test this code?

Added tests, tested manually

Table with comparison and informative tooltip on cell:
<img width="881" height="244" alt="image" src="https://github.com/user-attachments/assets/a9bc792a-f050-4149-9813-8660dfffb003" />

Commit:
feat(marketing-analytics): add compareFilter for date range comparisons in queries

- Added  with description to schema and types for better clarity.
- Implemented a generic cell renderer for marketing analytics that formats current and previous period values.
- Updated query runner to handle comparisons between current and previous periods, including necessary adjustments in the query building process.
- Removed unused number formatting logic to streamline rendering.

Commit:
refactor(marketing-analytics): extract renderMarketingAnalyticsCell to shared module

- Moved the generic cell renderer for marketing analytics from MarketingAnalyticsTable to shared.tsx for better reusability.
- Removed the inline implementation in MarketingAnalyticsTable to streamline the component.
- Updated imports accordingly to maintain functionality.

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

Commit:
fixing test

Commit:
addind story

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-previous-comparison-on-table

Commit:
feat: enhance marketing analytics cell with tooltip and percentage change calculation

- Added Tooltip component to display current and previous values along with percentage change.
- Implemented calculatePercentageChange function to handle percentage calculations.
- Updated renderMarketingAnalyticsCell to format values and handle null cases more effectively.

Commit:
Adding more css improving the UI

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-previous-comparison-on-table

Commit:
fixing css

Commit:
simplifying logic

Commit:
conflict merge

Commit:
fixing python baseline filter

Commit:
Adding new test suit for table comparison

Commit:
fixing failing test

jabahamondes commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/MarketingAnalyticsTable/MarketingAnalyticsTable.tsx:
```diff
@@ -22,11 +28,28 @@ export const MarketingAnalyticsTable = ({ query, insightProps }: MarketingAnalyt
     const { showColumnConfigModal } = useActions(marketingAnalyticsLogic)
 
+    // Filter out columns that are not numbers because they can't be compared between periods
+    const nonNumberColumns = Object.keys(MARKETING_ANALYTICS_SCHEMA).filter(
+        (column) => !MARKETING_ANALYTICS_SCHEMA[column as MarketingAnalyticsColumnsSchemaNames].type.includes('number')
+    )
```
it's typed on MarketingAnalyticsSchemaFieldTypes

Commit:
Prs comments

jabahamondes commented on frontend/src/queries/nodes/DataTable/renderColumn.tsx:
```diff
@@ -368,10 +373,5 @@ export function renderColumn(
     }
 
-    // Add number formatting for numeric values
-    if (context?.formatNumbers && typeof value === 'number') {
```
(i'm added this code and was the only one using it)

Commit:
removing unused code

jabahamondes commented:
FYI I will add the color depending increase/decrease in a following PR bc i would involve some refactoring

lricoy reviewed (approved):
LGTM 🚀 Looking forward for the next one with the color and extra changes

Commit:
Refactor marketing analytics tooltip and change calculation

- Removed the  function and integrated its logic directly into .
- Updated  to accept  as a parameter for better clarity.
- Simplified the logic in  to handle null values more gracefully.
- Adjusted the tooltip to display the formatted change percentage instead of the previous value.

This streamlines the code and improves readability while maintaining functionality.

Commit:
Refactor marketing analytics cell rendering

- Renamed  to  for better clarity and consistency.
- Updated import statements in  to reflect the new function name.
- Adjusted the rendering logic to directly use the renamed function.

This change enhances code readability and maintains functionality.
